subprojects/launcher/src/main/java/org/gradle/launcher/cli/ExceptionReportingAction.java

```diff
@@ -42,6 +45,9 @@ public void execute(ExecutionListener executionListener) {
         } catch (ReportedException e) {
             // Exception has already been reported
             executionListener.onFailure(e);
+        } catch (ServiceCreationException e) {
+            reporter.execute(new ContextAwareException(new InitializationException(e)));
```

Comment:
The current error handling framework expects a `LocationAwareException` in order to parse the call stack and construct the messages based on `@Contextual` annotations. I did not want to create a `LocationAwareException` here with null source and null line, thus extracted the code dealing with `@Contextual` into `ContextAwareException` that is now a superclass of `LocationAwareException`.